docs/slide-in-use-cases.md

```diff
@@ -0,0 +1,24 @@
+# Overview through Slide In use cases
+
+This document is an overview through the use cases.
+It should help to implement and understand the feature.
+
+## Use cases
+
+| UC ID | Local branch exists | Branch entry exists | Estimated frequency |
+| :---: | :---: | :---: | :---: |
+| 1 | false | false | ~ 75% |
+| 2 | false | true  | < 1%  |
+| 3 | true  | false | ~ 25% |
+| 4 | true  | true  | < 1%  |
+
+## Slide In - Logic
+
+This specification covers all of the cases.
+
+- When the **entry does not exist**, it must be added under the given branch.
+- When the **entry exists under the given parent**, branch layout must not be affected.
+- When the **entry exists under other parent or is root**, it must be reattached under the given parent.
```

Comment:
I think that some kind of question dialogbox will be nice in this case - this is something that (esp. beginning users) can overlook and this potentially can destroy branches tree.

Reply:
have you seen the slide in dialog? I think you might have omitted it.
![image](https://user-images.githubusercontent.com/19799111/87808613-506aa680-c85a-11ea-9867-97b27ce56291.png)

It clearly indicates what is to be slide in below what. 
Even though the newbie misses the **bold** parent he/she can easily undo things within (conveniently accessible) machete file. 

Adding the next dialog `Are you sure that you are sure?` is unnecessary.

Reply:
+ actually the comment about dialog name applies here

Reply:
Yes, I have seen this dialog, but IMHO i someone will try slide-in branch that is present under other parent this is not something fully intuitive that this branch will be reattached. I mean (esp. in large branch tree) this is simple to accidentally omit already added branch and trying to slide-in it again. But ok, I do not insist, maybe it's just my misconception that the user should be treated as a potential idiot xD

Reply:
I agree with your **first experience**.

Probably our different expectations (it was @PawelLipski suggestion to reattach) should be considered... :thinking: 
My idea is to add a checkbox (may be enabled by default) for the reattach (simply slide out and slide in without children otherwise). 

This is a candidate for another issue, WDYT?

Reply:
Yes, checkbox is definitely good idea (maybe even better than messagebox) and yes, this can be candidate for other issue

Reply:
issue #439  created